services/app-api/src/authn/authn.ts

```diff
@@ -4,5 +4,6 @@ import type { Store } from '../postgres'
 
 export type userFromAuthProvider = (
     authProvider: string,
-    store?: Store
-) => Promise<Result<UserType, Error>>
+    store?: Store,
+    userId?: string
+) => Promise<Result<UserType, Error | undefined>>
```

Comment:
We decided not to return undefined, right?

Reply:
and, this is confusing b/c it's the only place in the codebase we're using Result, but I think the right way to write this would be `Result<UserType, Error, undefined>` as opposed to using the sum type